hearth/lib/hearth/client.rb

```diff
@@ -38,7 +38,7 @@ def initialize_config(options, config_class)
       config.plugins.each { |p| p.call(config) }
       config.interceptors.concat(client_interceptors)
       config.validate!
-      config.freeze
+      config
```

Comment:
Why not freeze it here?  Only the plugins should be able to modify this and once they've done that, it should be frozen I think?

Reply:
I think maybe Config should be mutable once it's been assigned to the client, as an additional way to change config? We do this in v3. I'm on the fence about it. There are probably situations where we want to do this.

Reply:
Mutation of config on the client is not thread safe so should be avoided - I think we should not allow it.  Additionally, I think we want to make sure that interceptors are not able to modify it.

Reply:
It's true it's not thread safe. That doesn't necessarily mean we shouldn't do it. We will probably need to freeze config twice if that's the case?

Reply:
By twice you mean in two places (client initialize and operation invocation) right?  That seems correct.